mailpoet/tests/integration/REST/Automation/Workflows/WorkflowsGetTest.php

```diff
@@ -24,4 +47,133 @@ public function testRequest(): void {
+      $result = $this->get(self::ENDPOINT_PATH, ['query'=> ['status' => $status]]);
+      $this->assertCount(1, $result['data']);
+      $this->assertEquals($status, $result['data'][0]['name']);
+      $this->assertEquals($status, $result['data'][0]['status']);
+    }
+  }
+
+  /**
+   * This small helper method can quickly assert strings or integers from
+   * the REST API.
+   */
+  private function assertWorkflowRestData($expectation, $data) {
+    // We do not expect steps.
+    $this->assertFalse(isset($data['steps']));
+    unset($expectation['steps']);
+
+    foreach ($expectation as $key => $expectedValue) {
+      $this->assertEquals($expectedValue, $data[$key], "Failed asserting that the property $key is equal.");
+    }
+
+    // Check activated_at behavior
+    if (isset($expectation['status']) && $expectation['status'] !== Workflow::STATUS_DRAFT) {
+      $this->assertNotNull($data['activated_at']);
+      //Check we can read the date correctly.
+      $date = null;
+      try {
+        $date = new \DateTimeImmutable($data['activated_at']);
+      } finally {
+        $this->assertInstanceOf(\DateTimeImmutable::class, $date);
+      }
+    }
+    if (isset($expectation['status']) && $expectation['status'] === Workflow::STATUS_DRAFT) {
+      $this->assertNull($data['activated_at']);
+    }
+
+    // Check the date time fields are convertible.
+    $updatedAt = null;
+    $createdAt = null;
+    try {
+      $updatedAt = new \DateTimeImmutable($data['updated_at']);
+      $createdAt = new \DateTimeImmutable($data['created_at']);
+    } finally {
+      $this->assertInstanceOf(\DateTimeImmutable::class, $updatedAt);
+      $this->assertInstanceOf(\DateTimeImmutable::class, $createdAt);
+    }
+  }
+
+  private function createNewWorkflow(array $data = []) : int {
+    $rootStep = ['id'=>'root','type'=>Step::TYPE_ROOT,'key'=>'core:root'];
+    $data['name'] = $data['name']??'Test';
+    $data['steps'] = $data['steps']??[$rootStep];
+    $data['author'] = $data['author']??wp_get_current_user()->ID;
+    $workflow = new Workflow(
+      $data['name'],
+      array_map([$this,'createStep'],$data['steps']),
+      new \WP_User((int)$data['author'])
+    );
+    $workflow->setStatus($data['status']??Workflow::STATUS_ACTIVE);
+    return $this->workflowStorage->createWorkflow($workflow);
+  }
+
+  private function createStep(array $data = []) : Step {
+    $data['id'] = $data['id']??uniqid();
+    $data['type'] = $data['type']??Step::TYPE_ACTION;
+    $data['key'] = $data['key']??'key';
+    $data['args'] = $data['args']??[];
+    $data['nextSteps']=$data['nextSteps']??[];
+    return new Step(
+      $data['id'],
+      $data['type'],
+      $data['key'],
+      $data['args'],
+      $data['nextSteps']
+    );
+  }
+
+  public function _after() {
+    $this->workflowStorage->flush();
+    foreach ($this->userIds as $userId) {
+      is_multisite() ? wpmu_delete_user($userId) : wp_delete_user($userId);
+    }
+    parent::_after();
+  }
```

Comment:
I see some whitespacing issues here but I think that's due to https://a8c.slack.com/archives/C01GH835U90/p1663847701224009 so let's keep it and fix it there...